components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java

```diff
@@ -696,6 +700,34 @@ private String getCaptchaParams(String tenantDomain) {
                                 log.debug("Enforcing recaptcha for resend confirmation is not enabled.");
                             }
                         }
+                    } else if (captchaEnabledConfigName.equals(connectorConfig.getName())) {
+                        if (Boolean.parseBoolean(connectorConfig.getValue())) {
+                            Property[] maxFailedConfig = BasicAuthenticatorDataHolder.getInstance()
+                                    .getIdentityGovernanceService()
+                                    .getConfiguration(new String[]{maxFailedAttemptCaptchaConfigName}, tenantDomain);
+                            Property maxFailedProperty = maxFailedConfig[0];
+                            try {
+                                if (Integer.valueOf(maxFailedProperty.getValue()) < failedLoginAttempts) {
+                                    if (log.isDebugEnabled()) {
+                                        log.debug("Number of failed attempts is higher than max failed " +
+                                                "attempts for reCaptcha. Recaptcha will be enforced");
+                                    }
+                                    captchaParams += BasicAuthenticatorConstants.RECAPTCHA_PARAM + "true";
+                                } else {
+                                    if (log.isDebugEnabled()) {
+                                        log.debug("Number of failed attempts is less than max failed " +
+                                                "attempts for reCaptcha. Recaptcha will not be enforced");
+                                    }
+                                }
+                            } catch (NumberFormatException e) {
+                                log.error(String.format("Invalid Max failed attempts for reCaptcha: %s",
+                                        maxFailedProperty.getValue()), e);
+                            }
+                        } else {
+                            if (log.isDebugEnabled()) {
+                                log.debug("Enforcing recaptcha for SSO Login is not enabled.");
+                            }
```

Comment:
Can't we just end the flow if not enabled?

Reply:
Same case as in above comments